addon/index.js

```diff
@@ -1,144 +1,15 @@
 import Ember from 'ember';
+import StatefulMixin from 'ember-stateful/mixins/stateful';
 
-const { computed } = Ember;
-
-/**
- * @see https://github.com/emberjs/ember.js/blob/v2.14.1/packages/ember-views/lib/mixins/action_support.js
- */
-export default Ember.Mixin.create({
-
-  /**
-   * you need to override this
-   */
-  states: undefined,
-
-  currentState: computed({
-    get() {
-      return this._getTargetState('');
-    },
-    set(key, value) {
-      return this._getTargetState(value);
-    },
-  }),
-
-  /**
-   * If the current State is 'a.b.c', this property is `{ a: { b: { c: {}}}}`
-   * The idea is to simplify state checks, e.g.
-   * `if (this.get('state.a.b')) { ... }` will work in state 'a.b.c' and 'a.b.d',
-   * epsecially for templates.
-   *
-   * @property {Object} state
-   */
-  state: computed('currentState', function() {
-    return this.get('currentState').split('.').reduceRight((acc, s) => {
-      return { [s]: acc };
-    }, {});
-  }),
-
-  _defaultStateMapping: undefined,
-
-  _getTargetState(value) {
-    let targetState = value;
-    do {
-      value = targetState;
-      targetState = this._defaultStateMapping[value];
-    } while (!Ember.isNone(targetState) && targetState !== value)
-    if (Ember.isNone(targetState)) {
-      throw new Error(`Invalid state transition attempted. Unknown state: ${value}`);
-    }
-    return targetState;
-  },
-
-  init(...args) {
-    this._super(...args);
-    const states = this.get('states');
-    // put some empty actions into the actions hash so components don't complain
-    const newRootStateActions = {};
-    const stateToDefaultStateMap = {};
-    // process in reverse order, because default states will be listed first
-    states.reverse().forEach(state => {
-      let stateParts = state.split('.');
-      if (Ember.isNone(stateToDefaultStateMap[state])) {
-        stateToDefaultStateMap[state] = state;
-        // if stateToDefaultStateMap[state] is already set, the consumer
-        // used a redundant intermediate state
+export function waitForEnterState(statefulObject, stateName) {
```

Comment:
`waitForState` which resolves immediately if we're already in the given state or waits to enter it would be a more useful helper.

Reply:
Well, you can always check that explicitly, and this could be useful for cases where there's a transition that comes back to the start state. I'll add `waitForState()` too though.